From b2cee2d290c533d911e17084f5013dc7a09a1cc3 Mon Sep 17 00:00:00 2001
From: Pantelis Antoniou <panto@antoniou-consulting.com>
Date: Tue, 22 Jan 2013 22:25:06 +0200
Subject: [PATCH 09/11] MUSB: Hack around to make host port to work.

It sort of works, but it doesn't handle hotplug.
After removing the device, issue lsusb to cause it to scan the bus again.

Crappy, I know, but at least works.

Merged by Jason Kridner <jdk@ti.com> against updated musb code
---
 arch/arm/boot/dts/am335x-bone-common.dtsi |    6 ++++++
 arch/arm/boot/dts/am33xx.dtsi             |    5 +++--
 drivers/usb/musb/musb_core.c              |    2 ++
 drivers/usb/musb/musb_host.c              |    4 ++++
 4 files changed, 15 insertions(+), 2 deletions(-)

diff --git a/arch/arm/boot/dts/am335x-bone-common.dtsi b/arch/arm/boot/dts/am335x-bone-common.dtsi
index 524bf03..fc711df 100644
--- a/arch/arm/boot/dts/am335x-bone-common.dtsi
+++ b/arch/arm/boot/dts/am335x-bone-common.dtsi
@@ -449,3 +449,9 @@
 &aes {
 	status = "okay";
 };
+
+&usb_otg_hs {
+	interface_type = <1>;
+	power = <500>;
+	status = "okay";
+};
diff --git a/arch/arm/boot/dts/am33xx.dtsi b/arch/arm/boot/dts/am33xx.dtsi
index 073eb36..49049a4 100644
--- a/arch/arm/boot/dts/am33xx.dtsi
+++ b/arch/arm/boot/dts/am33xx.dtsi
@@ -270,11 +270,11 @@
 			status = "disabled";
 		};
 
-		usb0_phy: phy0 {
+		usb0_phy: nop-phy@0 {
 			compatible = "nop-xceiv-usb";
 		};
 
-		usb1_phy: phy1 {
+		usb1_phy: nop-phy@1 {
 			compatible = "nop-xceiv-usb";
 		};
 
@@ -289,6 +289,7 @@
 			power = <250>;
 			usb0-phy = <&usb0_phy>;
 			usb1-phy = <&usb1_phy>;
+			status = "disabled";
 		};
 
 		spi0: spi@48030000 {
diff --git a/drivers/usb/musb/musb_core.c b/drivers/usb/musb/musb_core.c
index 9ad3c39..591b86c 100644
--- a/drivers/usb/musb/musb_core.c
+++ b/drivers/usb/musb/musb_core.c
@@ -1991,6 +1991,8 @@ musb_init_controller(struct device *dev, int nIrq, void __iomem *ctrl)
 		musb_write_ulpi_buscontrol(musb->mregs, busctl);
 	}
 
+	/* TODO/FIXME: figure out if it is sane to have this configured 
+	 * even if we have ports that are host-only */
 	MUSB_DEV_MODE(musb);
 	musb->xceiv->otg->default_a = 0;
 	musb->xceiv->state = OTG_STATE_B_IDLE;
diff --git a/drivers/usb/musb/musb_host.c b/drivers/usb/musb/musb_host.c
index 112900b..4d527a2 100644
--- a/drivers/usb/musb/musb_host.c
+++ b/drivers/usb/musb/musb_host.c
@@ -2531,12 +2531,16 @@ static int musb_bus_suspend(struct usb_hcd *hcd)
 		break;
 	}
 
+#if 0
 	if (musb->is_active) {
 		WARNING("trying to suspend as %s while active\n",
 				otg_state_string(musb->xceiv->state));
 		return -EBUSY;
 	} else
 		return 0;
+#else
+	return 0;
+#endif
 }
 
 static int musb_bus_resume(struct usb_hcd *hcd)
-- 
1.7.8.6

